core/src/Kokkos_View.hpp

```diff
@@ -1725,6 +1752,48 @@ class View : public ViewTraits<DataType, Properties...> {
                   "Layout is not constructible from extent arguments. Use "
                   "overload taking a layout object instead.");
   }
+
+  //----------------------------------------
+  // MDSpan converting constructors
+#ifdef KOKKOS_ENABLE_IMPL_MDSPAN
+  template <typename U = typename Impl::MDSpanViewTraits<traits>::mdspan_type>
+  KOKKOS_INLINE_FUNCTION MDSPAN_CONDITIONAL_EXPLICIT(traits::is_managed) View(
```

Comment:
`MDSPAN_CONDITIONAL_EXPLICIT` will not be defined if we use an implementation that if not the Kokkos one.  You need to guard the use of the explicit specifier with an expression against `KOKKOS_ENABLE_CXX20`

Reply:
Let us do KOKKOS_IMPL_CONDITIONAL_EXPLICIT